backend/devices/bitbox/relay/channel.go

```diff
@@ -148,13 +148,13 @@ func (channel *Channel) waitForValue(duration time.Duration, name string) (strin
 			unlock()
 			message, err := PullOldestMessage(relayServer(), channel)
 			if err != nil {
-				return "", err
+				return "pullFailed", err
```

Comment:
By convention, if a function returns an error, all other response values should be treated as undefined and not inspected.

In this case, it is a better fit to return `"", ErrPullFailed` or similar,  have the caller of the function check against this.

After all, you are not receiving the "pullFailed" value from the sever.